### PR TITLE
[CDAP-7706] Initial unit tests for Schema Editor

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Ports `avsc` library from plain flat library to npm module
- Removes existing avsc library bundles.
- Makes appropriate changes to both hydrator & cdap to use the common `cdap-avsc`
- Fixes `.stylelintrc` to avoid the warning message
- Adds initial unit tests (basic nothing substantial) to SchemaEditor to kick off unit testing for the same.

Bamboo build: http://builds.cask.co/browse/CDAP-DRC4924-5

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/detail/top-panel-ctrl.js [hunk 1]
```diff
@@ -138,7 +138,7 @@ class HydratorDetailTopPanelController {
     let res = true;
 
     for(let key in this.macrosMap){
-      if(!this.macrosMap[key]){
+      if (this.macrosMap.hasOwnProperty(key) && !this.macrosMap[key]) {
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/controllers/detail/top-panel-ctrl.js:141]: so we need the hasOwnProperty?

File: cdap-ui/.stylelintrc [hunk 2]
```diff
@@ -58,7 +58,6 @@
     "no-missing-eof-newline": true,
     "number-leading-zero": "always",
     "number-no-trailing-zeros": true,
-    "number-zero-length-no-unit": true,
```
Comment (elfenheart) [inline cdap-ui/.stylelintrc:61]: add `length-zero-no-unit` ?

File: cdap-ui/app/hydrator/controllers/detail/top-panel-ctrl.js (hunk 1 above)
Comment (elfenheart) [inline cdap-ui/app/hydrator/controllers/detail/top-panel-ctrl.js:141]: so we need `hasOwnProperty`?

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/detail/top-panel-ctrl.js:141]: Yup. Gulp was failing.

File: cdap-ui/.stylelintrc (hunk 2 above)
Comment (ajainarayanan) [reply cdap-ui/.stylelintrc:61]: This was a warning from stylelint gulp plugin. We had a deprecated warning.

Comment (elfenheart) [reply cdap-ui/.stylelintrc:61]: yeah, but it says to replace with `length-zero-no-unit` right?

Comment (ajainarayanan) [reply cdap-ui/.stylelintrc:61]: Fixed.

**Comment by elfenheart** [comment]
👍  LGTM

**Pull request merged**